test/test-threadpool-stats.c

```diff
@@ -0,0 +1,87 @@
+/* Copyright libuv project contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+#include "test-threadpool.h"
+
+static int submitted;
+static int started;
+static int done;
+#define START_SIZE 8
+static int idle_max;
+static int idle_min = 2 * START_SIZE;
+static int queued_max;
+
+static void update(int queued, int idle) {
+  if (queued > queued_max) queued_max = queued;
+  if (idle > idle_max) idle_max = idle;
+  if (idle < idle_min) idle_min = idle;
+}
+
+static void stats_submit_cb(int queued, int idle, void* data) {
+  fprintf(stderr, "submit: q %d i %d\n", queued, idle);
```

Comment:
As a rule of thumb, tests should be silent (i.e., no printf.)

Reply:
Are you sure? The current tests use lots of `fprintf(stderr` - and the test runner does the right thing, it suppresses all output unless the test fails, then it shows the output so the test failure can be debugged. I don't mind deleting it, but it looked to me like this was the pattern in libuv's tests.